metric/unaggregated/types.go

```diff
@@ -119,14 +123,14 @@ func (m *MetricUnion) String() string {
 	}
 }
 
-// Reset resets the metric union
+// Reset resets the metric union.
```

Comment:
This might be tricky because imagine a scenario where a timer is decoded and the value is being processed/has been processed and returned to pool, and now the iterator is reset/closed due to broken connection etc, during which the metric union is reset. If we return the floats to the pool, we may risk returning the values twice.

I'd rather err on the side of safety and not return it to the pool, given how rare this is. The pool also has logic that if the pool level is below certain watermark, it'll automatically refill. The longer-term solution will be make the values refcounted so we don't need to explicitly track ownership (see my todo above the `MetricUnion` definition).